src/mixins/viewport/ViewportMixin.js

```diff
@@ -49,20 +49,30 @@ var Viewport = {
         } else {
             margins = viewportMargins;
         }
+        
         // Detect Screen Bottom                           // Detect Screen Top
         if ((rect.top < window.innerHeight + margins.top) && (rect.bottom > 0  - margins.bottom)) {
-            if (!self.isOnViewport) {
-                self.enterViewportCallback && self.enterViewportCallback();
-                self.isOnViewport = true;
+            if (!i13nNode.isInViewport()) {
+                enterViewportCallback && enterViewportCallback();
+                i13nNode.setIsInViewport(true);
             }
         } else {
-            if (self.isOnViewport) {
-                self.exitViewportCallback && self.exitViewportCallback();
-                self.isOnViewport = false;
+            if (i13nNode.isInViewport()) {
+                exitViewportCallback && exitViewportCallback();
+                i13nNode.setIsInViewport(false);
             }
         }
     },
 
+    _detectViewport: function () {
+        var self = this;
+        if (!self.isMounted()) {
+            return;
+        }
+        self._detectElement(self._i13nNode, self.enterViewportCallback, self.exitViewportCallback);
+        self._subComponentsViewportDetection && self._subComponentsViewportDetection();
```

Comment:
change some detail implementation of viewport detection, will detect subcomponents as well